RedditSharp/Things/Subreddit.cs

```diff
@@ -508,12 +508,7 @@ public Listing<PrivateMessage> GetModmail()
 
         private void SetName()
         {
-            Name = Url.ToString();
-            if (Name.StartsWith("/r/"))
-                Name = Name.Substring(3);
-            if (Name.StartsWith("r/"))
-                Name = Name.Substring(2);
-            Name = Name.TrimEnd('/');
+            Name = System.Text.RegularExpressions.Regex.Replace(Url.ToString(), "(r/|/)", "");
```

Comment:
This will fail in certain situations

"/r/aaaaar/" => "aaaaa" != "aaaaar"

Regex needs to specify only replacing the "r/" at the beginning of a string